ndb/MIGRATION_NOTES.md

```diff
@@ -92,3 +94,10 @@ The primary differences come from:
   a new `Property()` instance is created. This increment is not threadsafe.
   However, `ndb` was designed for `Property()` instances to be created at
   import time, so this may not be an issue.
+- Since we want "compatibility", suggestions in `TODO` comments have not been
+  implemented. However, that policy can be changed if desired.
+- It seems that `query.ConjunctionNode.__new__` had an unreachable line
+  that returned a `FalseNode`. This return has been changed to a
+  `RuntimeError` just it case it **is** actually reached.
+- For ``AND`` and ``OR`` to compare equal, the nodes must come in the
+  same order. So ``AND(a > 7, b > 6)`` is not equal to ``AND(b > 6, a > 7)``.
```

Comment:
Are AND and OR not commutative because of logic or because of history?

Reply:
Because of history.

Reply:
I can't imagine it would break anything to fix that, then.

Reply:
Sure thing, will do.

Reply:
OK, scratch that. It's currently not possible:

- `sorted(self._nodes)` fails because the `Node` types are **explicitly** not comparable
- `set(self._nodes)` fails because the `Node` types are not hashable (which @tseaver is commenting on below)

WDYT is the right move here?

Reply:
Maybe, like __hash__, it's YAGNI.  I wonder how often these are tested for equality, anyway.  Almost never, maybe?

Reply:
Yes I'd imagine given the age of the library, it's not a feature worth adding / supporting.